oyoyo/client.py

```diff
@@ -174,7 +180,60 @@ def connect(self):
                         sys.exit(1)
 
             if self.use_ssl:
-                self.socket = ssl.wrap_socket(self.socket)
+                ctx = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+
+                if self.cert_verify or self.cert_fp:
+                    ctx.verify_mode = ssl.CERT_REQUIRED
+                    ctx.load_default_certs()
+
+                if self.client_certfile:
+                    # if client_keyfile is not specified, the ssl module will look to the
+                    # client_certfile for it.
+                    ctx.load_cert_chain(self.client_certfile, self.client_keyfile)
+
+                try:
+                    self.socket = ctx.wrap_socket(self.socket)
+                except (ssl.CertificateError, ssl.SSLError, Exception) as error:
+                    print("Error occured while connecting with TLS: {0}".format(error))
```

Comment:
Please use `self.stream_handler` instead of `print` as it has more useful logging.